src/sonic-eventd/rsyslog_plugin/syslog_parser.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <ios>
+#include <fstream>
+#include <regex>
+#include "syslog_parser.h"
+#include "common/logger.h"
+
+
+using namespace std;
+/**
+ * Parses syslog message and returns structured event
+ *
+ * @param nessage us syslog message being fed in by rsyslog.d
+ * @return return structured event json for publishing
+ *
+*/
+
+bool SyslogParser::parseMessage(string message, string& event_tag, event_params_t& param_map) {
+    for(long unsigned int i = 0; i < regex_list.size(); i++) {
+        smatch match_results;
+	regex_search(message, match_results, expressions[i]);
```

Comment:
We can skip the rest, if regex_search returns false and add a continue statement to go to next expression.

Reply:
Done.